### ISSUE TITLE
Can I use the templates in asset's path config?

### ISSUE THREAD

**Issue by lfilho** [semantic-release/github#274]
Hi there, thanks for the great project!

I have the following files:

```
dist/file1-v1.0.0.zip
dist/file1-v1.0.1.zip
dist/file1-v1.1.0.zip
...etc
```

Then I **had** this configuration:

```
      [
        "@semantic-release/github",
        {
          "assets": "dist/*.zip"
        }
      ]
```
With the config above, all my files in `dist/` folder were getting attached to each release in Github's release page. So far so good, it was working as expected and it was an overlook from my end.

Then I tried to fix it by doing the following:

```
      [
        "@semantic-release/github",
        {
          "assets": "dist/*${nextRelease.version}.zip"
        }
      ]
```

But it didn't work -- no assets were attached to the next release. Am I missing something? Are templates even supported in `path`?

Thank you!

**Comment by gr2m** [comment]
I don't know out of my head. But I think @coderbyheart implemented the file templates feature. Maybe he can help?

**Comment by coderbyheart** [comment]
No, you can't use the template strings when resolving files, because they are not replaced and passed verbatim: https://github.com/semantic-release/github/blob/32654fb26dbba597170408ab181255ae863ed748/lib/publish.js#L51-L60

I recommend to store assets with an unversioned name define individual asset entries, with a name property that contains the desired name.

Here is an example of how I use it in a project: https://github.com/bifravst/firmware/blob/8632a44c5201c24c554c06861cbb32db11682b26/package.json#L44-L90

**Comment by levibostian** [comment]
Thanks for the help @coderbyheart. Your code snippet + writing a script that copies all of my generated files into hard coded paths worked!

**Comment by jedwards1211** [comment]
`javapackager` appends the project version to the base output name I choose, and it doesn't seem possible to disable this, so it would be nice to have the flexibility to use template strings in asset paths.

**Comment by jedwards1211** [comment]
@gr2m would you be opposed to a PR to also call `template` on the asset paths before actually globbing them?

**Comment by jedwards1211** [comment]
Or I guess making an unversioned symlink to the versioned filename would be palatable. I don't get the reasoning behind intentionally not supporting parameterized filenames though.  Seems unlikely that someone would have `${foo}` in a filename and need to match it verbatim.  And if so, not hard to escape the `$`/not hard to explain this in the docs.

**Comment by levibostian** [comment]
@jedwards1211 if it helps, here is what I did to get it working in the mean time. 

Follow the [advice](https://github.com/bifravst/firmware/blob/8632a44c5201c24c554c06861cbb32db11682b26/package.json#L44-L90) from @coderbyheart and add `path`, `label`, and `name` properties. The `name` can add the version number. 

Rename the file that `javapacker` produced to a static file name. Reference the hard coded path in `path` of the assets object.

**Comment by molvqingtai** [comment]
Why should it be consistent with Gitlab's implementation? If the intention is to not support templates for paths, it should at least be mentioned in the documentation, which has cost me a whole day.

https://github.com/semantic-release/gitlab/pull/630/files
